File: src/main/scala/com/kirkwoodwest/scalawoods/utils/LogUtil.scala
package com.kirkwoodwest.scalawoods.utils

import com.bitwig.extension.controller.ControllerExtension
import com.bitwig.extension.controller.api.ControllerHost

/**
 * Simple Log tool to initialized in the main extension and then used in any class.
 */
object LogUtil {
  private var _host: ControllerHost = null

  
  def init(host: ControllerHost): Unit = {
    _host = host
  }

  def print(s: String): Unit = {
    _host.println(s)
  }

  def getHost: ControllerHost = _host
  
  def println(s: String): Unit = {
    _host.println(s)
  }

  def h1(s: String): Unit = {
    _host.println("--- " + s.toUpperCase + "--------------------------")
  }

  def indent(s: String, indent: Int): Unit = {
    var s2 = ""
    for (i <- 0 until indent) {
      s2 = s2 + "-"
    }
    s2 = s2 + " " + s
    _host.println(s2)
  }

  def reportExtensionStatus(extension: ControllerExtension, status_message: String): Unit = {
    //If your reading this... Say hello to a loved one today. <3
    val name = extension.getExtensionDefinition.getName
    val version = extension.getExtensionDefinition.getVersion
    _host.println(name + " " + version + " " + status_message)
  }
}

File: src/main/java/com/kirkwoodwest/openwoods/utils/LogUtil.java
package com.kirkwoodwest.openwoods.utils;

import com.bitwig.extension.controller.ControllerExtension;
import com.bitwig.extension.controller.api.ControllerHost;

/**
 * Simple Log tool to initialized in the main extension and then used in any class.
 */
public class LogUtil {
  private static ControllerHost _host = null;
  public static void init(ControllerHost host){
    _host = host;
  }
  public static void print(String s){

    _host.println(s);
  }
  public static void println(String s){
    _host.println(s);
  }

	public static void h1(String s) {
    _host.println("--- " + s.toUpperCase() + "--------------------------");
	}

	public static void indent(String s, int indent) {
    String s2 = "";
    for(int i=0;i<indent;i++){
      s2 = s2 + "-";
    }
    s2 = s2 + " " + s;
    _host.println(s2);
  }

  public static ControllerHost getHost(){
    return _host;
  }


  public static void reportExtensionStatus(ControllerExtension extension, String status_message) {
    //If your reading this... Say hello to a loved one today. <3
    String name = extension.getExtensionDefinition().getName();
    String version = extension.getExtensionDefinition().getVersion();
    _host.println(name + " " + version + " " + status_message);
  }

}
